staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -87,6 +92,10 @@ func (s *store) RequestWatchProgress(ctx context.Context) error {
 	return s.client.RequestProgress(s.watchContext(ctx))
 }
 
+func (s *store) Supports(ctx context.Context, feature storage.Feature) (bool, error) {
```

Comment:
@serathius - I actually started wondering here - we can easily use this function to check if we can use ConsistentReads feature.

But we also need to use that information to warn about bad version now (and eventually fail the kube-apiserver) early in the cycle - not only on the first call.
How we're going to do that?

Reply:
This is a watch cache feature, so around the time when watch cache is created? https://github.com/kubernetes/kubernetes/blob/fc7325fbec0fe379694e58b82e09f99f166c5e6d/staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go#L335

We get the storage and can check the flag. If they don't match we can warn or return error from there.

This reminded me about more problematic part, estimations of request cost. Enabling the feature changes the cost estimations of list requests. So if feature flag is enabled, but the etcd is too old and consistent list from cache is disabled it means that cluster will get too optimistic cost estimations. This would be a problem for APF.

I think we need to reconsider having one global storage feature flag, so that cost estimation function can check it.